cores/esp8266/PolledTimeout.h

```diff
@@ -211,11 +219,11 @@ class timeoutTemplate
 private:
 
   IRAM_ATTR // fast
-  bool checkExpired(const timeType internalUnit) const
+  bool checkExpired() const
```

Comment:
I'm completely at ease with the rename.

Reply:
I think there are various problems with the auto-reset timeout, as well as the one-shot timeout. Btw, expiredOneShot() is incorrect. Consider staying a polling loop beyond the timeType's relative overflow. Boom.

Reply:
Since it's now back to the basic addition that I need - asking the remaining time until next timeout, I'll rather not like to make any further changes, in hopes that this can be merged without much concern.
I hope you are alright with the fix to one-shot, though.

Reply:
@devyte thankfully the CI caught how I had broken `reset()` by overwriting `_timeout`. This is now fixed. I've permitted myself to introduce another field for the purpose of maintaining oneshot-has-expired until explicitly reset. Please check that particular change for any breakage you might suspect.